eq_translations/survey_schema.py

```diff
@@ -1,245 +1,205 @@
+        question = self.get_parent_question(pointer)
+
+        if "text_plural" in question:
+            message_context = question["text_plural"]["forms"]["other"]
+        elif "placeholders" in question:
+            message_context = question["text"]
+        else:
+            message_context = question
+
+        return f"Answer for: {message_context}"
+
+    @property
+    def catalog(self):
         """
         :return: a schema catalog to be used as the content within a po/pot file
         """
         catalog = Catalog()
-        total_translations = len(self.no_context_pointers + self.context_pointers)
 
-        for pointer in self.no_context_pointers:
-            pointer_contents = resolve_pointer(self.schema, pointer)
-            if pointer_contents:
-                catalog.add(dumb_to_smart_quotes(pointer_contents))
-
-        for pointer in self.context_pointers:
-            pointer_contents = resolve_pointer(self.schema, pointer)
-            if pointer_contents:
-                parent_answer_id = self.get_parent_id(pointer)
-                question = self.get_parent_question(pointer)
+        for pointer_dict in self.pointer_dicts:
+            message_context = pointer_dict.get("context")
+            pointer_contents = resolve_pointer(self.schema, pointer_dict["pointer"])
 
-                message_context = "Answer for: {}".format(
-                    question["text"] if is_placeholder(question) else question
-                )
+            if not pointer_contents:
+                continue
 
-                catalog.add(
-                    dumb_to_smart_quotes(pointer_contents),
-                    context=message_context,
-                    auto_comments=[f"answer-id: {parent_answer_id}"],
-                )
+            message_id = get_message_id(content=pointer_contents)
+            catalog.add(
+                id=message_id, context=message_context,
+            )
 
-        print(f"Total Messages: {total_translations}")
+        print(f"Total Messages: {len(self.pointer_dicts)}")
 
         return catalog
 
-    def translate(self, schema_translation):
+    def translate(self, schema_translation, language_code):
         """
-        Use the supplied schema translation object to translate pointers found
+        Use the supplied schema translation object and language code to translate pointers found
         within the survey
 
-        :param schema_translation:
+        :param schema_translation: The SchemaTranslation object
+        :param language_code: The target language code
         :return:
         """
         translated_schema = copy.deepcopy(self.schema)
-        total_translations = len(self.no_context_pointers + self.context_pointers)
         missing_translations = 0
 
-        for pointer in self.no_context_pointers:
+        for pointer_dict in self.pointer_dicts:
+            pointer = pointer_dict["pointer"]
             pointer_contents = resolve_pointer(self.schema, pointer)
```

Comment:
We already call resolve pointer when we're collecting the pointers and context. We should add the resolved value to the extracted objects, so we'd have something like:

```
[
  {
    "pointer": "/key1/key2",
    "value": "My translatable string",
    "context": "Answer for: ..."
  },
  ...
]
```

Then we don't need to resolve every pointer twice.

Reply:
Done